src/briefcase/integrations/android_sdk.py

```diff
@@ -440,6 +435,92 @@ def upgrade(self):
 """
             ) from e
 
+    def install_cmdline_tools(self) -> bool:
+        """Attempt to use 'latest' cmdline-tools to install the currently required
+        version of the Command-Line Tools.
+
+        The Briefcase-managed SDK should always have the required version of cmdline-
+        tools installed; however, user-provided SDKs may not have it.
+
+        :returns: True if successfully installed; False otherwise
+        """
+        self.tools.logger.info(
+            f"Installing Android Command-Line Tools {self.SDK_MANAGER_VER}...",
+            prefix=self.full_name,
+        )
+        self.tools.logger.info(f"Using Android SDK at {self.root_path}")
+        latest_sdkmanager_path = (
+            self.root_path
+            / "cmdline-tools"
+            / "latest"
+            / "bin"
+            / self.sdkmanager_filename
+        )
+        try:
+            self.tools.subprocess.run(
+                [
+                    latest_sdkmanager_path,
+                    f"cmdline-tools;{self.SDK_MANAGER_VER}",
+                ],
+                check=True,
+                stream_output=False,
+            )
+        except (OSError, subprocess.CalledProcessError) as e:
+            self.tools.logger.debug(str(e))
+            self.tools.logger.warning(
+                f"Failed to install cmdline-tools;{self.SDK_MANAGER_VER}"
+            )
+            return False
+        return True
+
+    def delete_legacy_sdk_tools(self):
+        """Delete any legacy Android SDK tools that are installed.
+
+        If no versions of the Command-Line Tools are installed but the 'tools' directory
+        exists, the legacy SDK Tools are probably installed. Since they have been
+        deprecated by more recent releases of SDK Manager, delete them and perform a
+        fresh install.
+
+        The Android SDK Tools were deprecated in Sept 2017.
```

Comment:
Is there a use case for this happening? AFAICT, we're only doing this check if it's Briefcase-managed tools (I wouldn't want to delete anything from a user-managed install); but Briefcase hasn't used the legacy SDK for well over a year. Is this a "just in case" cleanup, or is this cleaning up the side effect of the 26.1.1 API that you mention in a later comment? If it's the latter, is this a situation where we're going to be playing whack-a-mole, with Gradle installing legacy tools and us deleting them every time?

Reply:
I was only maintaining the existing behavior. I would imagine, though, this could be removed at this point. 

Although, it presence, at all, is made more interesting by the fact that the APK build process installs the old SDK Tools anyway.

Reply:
Ah - I missed that although this is a new method, it's a refactor of existing functionality.